### DISCUSSION TITLE
Problem receiving VALID response with 1 component fingerprint even though componentMatchingStrategy is set to MATCH_ALL

### DISCUSSION THREAD

**Discussion by ahnjs1987** [keygen-sh/community#163] [Q&A] [answered]
Multiple components fingerprints were registered to the machine object. If you attempt to authenticate the license key using one or more of these component fingerprints, a VALID response will be returned. Is there anything else I need to set up?

**Comment by ezekg** [comment]
Can you clarify your expected behavior here and share your use case? I'm not sure I'm following. It sounds like it's working as expected.

**Comment by ahnjs1987** [comment]
I was created a policy with parameters as below.
**"machineMatchingStrategy":"MATCH_ALL",
"requireFingerprintScope":true,
"requireComponentsScope":true,
"componentMatchingStrategy":"MATCH_ALL",**

I created license with that policy. and I created 1 machine object on that license. I created 2 machine components on machine.
And I tried to validate key with fingerprint and components fingerprints.
According to policy setting, I think all of the components fingerprint must be matched, then I can get an response as VALID.
Even if only one of the component fingerprint matches, a response is received with VALID. Is this correct?

**Reply by ezekg** [reply, marked as answer]
The assertion is that all of the provided fingerprints must match the some of machine's components, not that all of the machine's component fingerprints must be provided — i.e. you're saying "all of these fingerprints must exist", not "these fingerprints must be all that exists." To implement your desired logic, you should adjust your integration to always send 2 component fingerprints, even if the missing component is an empty string or null, if you're fingerprinting 2 components.

Hope that makes sense? Let me know.

**Reply by ahnjs1987** [reply]
I understood. So the "MATCH_ALL" means only "provided fingerprints must be matched with among stored fingerprints".

Thanks!!
